### ISSUE TITLE
IDE0059 False Positive

### ISSUE THREAD

**Issue by dalumedic** [dotnet/roslyn#77123]
**Version Used**: 
9.0.102
VS Code on Mac OS 15.3 with the latest C# and C# Dev Kit extensions.

**Steps to Reproduce**:

Simplified method exhibiting the behavior:

```
public static string ValidateIdSimple(string customerId)
{
    Lazy<StringBuilder> messageBuilder = new();
    var result = string.Empty;

    try
    {
        if (string.IsNullOrEmpty(customerId) || customerId.Length != 4)
            messageBuilder.Value.Append("Customer ID must be 4 characters.");
    }
    finally
    {
        if (messageBuilder.IsValueCreated)
            result = messageBuilder.ToString();
    }

    return result;
}
```

   The unnecessary assignment is here: `result = messageBuilder.ToString();`
   This is the only assignment of result outside of the initial value so it's clearly used in the return (when the StringBuilder has a value).

Simple Unit Tests (xunit - all passing)

```
public class CustomerServiceTests
{
    private const string DefaultError = "Customer ID must be 4 characters.";

    [Theory]
    [InlineData("", DefaultError)]
    [InlineData(null, DefaultError)]
    [InlineData("id_is_too_long", DefaultError)]
    [InlineData("cus1", "")]
    public void ValidateIdSimple_Should_Return_Expected_Message(string id, string expected)
    {
        var error = CustomerService.ValidateIdSimple(id);
        Assert.Equal(expected, error);
    }
}
```

**Diagnostic Id**:
No message is generated when calling dotnet build.

**Expected Behavior**:
IDE0059 should not be raised for the submitted code.

**Actual Behavior**:
Unnecessary assignment of a value to 'result'[IDE0059](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0059)
(local variable) string result
